.github/workflows/run_periodic_tests.yml

```
@@ -21,6 +25,9 @@ concurrency:
 jobs:
   run_unit_integration_and_coverage_tests:
     runs-on: ${{ matrix.os }}
+    permissions:
+      contents: read
+      checks: write  # Only if needed for test results
```

```suggestion
    permissions:
      contents: read
```

I think we should be fine without `checks: write` – unless you found that it's needed.